src/wallet/rpc/addresses.cpp

```diff
@@ -602,6 +603,7 @@ RPCHelpMan getaddressinfo()
     ret.pushKVs(detail);
 
     ret.pushKV("ischange", ScriptIsChange(*pwallet, scriptPubKey));
+    ret.pushKV("isactive", pwallet->IsDestinationActive(dest));
```

Comment:
ok, moving `isactive` next to `ismine`. You have a good point about legacy wallet deprecation. I suppose when that happens most of this new code will be deprecated as well since its mainly implemented only in `LegacyScriptPubKeyMan`. Would it make sense to only add the field to RPC response in legacy wallets?

Reply:
> I'm not sure if or when legacy wallet support is expected to be phased out, but if isactive is only different from ismine in legacy wallets, is it worth adding a field (and all the new methods), versus just updating the ismine field documentation in the help for descriptor wallets? (When I'm double-checking an address with getaddressinfo before using it to receive, ismine is a field I'm already checking.)

@jonatack after diving back in to this issue I realized that `ismine != isactive` for descriptor wallets all the time. I added tests in `wallet_keypool` to confirm this. The difference is whether or not the SPKM is ACTIVE (A descriptor wallet can have both active and inactive SPKMs and `ismine` at the RPC level merely indicates if the wallet can spend from the address).

So to summarize the test I just added:
- create descriptor wallet
- getnewaddress, getaddressinfo: ismine and isactive are both true
- encrypt the wallet
- getaddressinfo again: ismine is true but isactive is false